dimod/bqm/adjdictbqm.py

```diff
@@ -0,0 +1,301 @@
+
+        self._adj.setdefault(v, {v: self.dtype.type(0)})
+        return v
+
+    def get_linear(self, v):
+        """Get the linear bias of v.
+
+        Args:
+            v (hashable):
+                A variable in the binary quadratic model.
+
+        Returns:
+            float: The linear bias of v.
+
+        Raises:
+            ValueError: If v is not a variable in the binary quadratic model.
+
+        """
+        try:
+            return self._adj[v][v]
+        except KeyError:
+            pass
+        raise ValueError("{} is not a variable".format(v))
+
+    def get_quadratic(self, u, v):
+        """Get the quadratic bias of (u, v).
+
+        Args:
+            u (hashable):
+                A variable in the binary quadratic model.
+
+            v (hashable):
+                A variable in the binary quadratic model.
+
+        Returns:
+            float: The quadratic bias of (u, v).
+
+        Raises:
+            ValueError: If either u or v is not a variable in the binary
+            quadratic model, if u == v or if (u, v) is not an interaction in
+            the binary quadratic model.
+
+        """
+        if u == v:
+            raise ValueError("No interaction between {} and itself".format(u))
+        try:
+            return self._adj[u][v]
+        except KeyError:
+            pass
+        raise ValueError('No interaction between {} and {}'.format(u, v))
+
+    def iter_variables(self):
+        """Iterate over the variables of the binary quadratic model.
+
+        Yields:
+            hashable: A variable in the binary quadratic model.
+
+        """
+        return iter(self._adj)
+
+    def pop_variable(self):
+        """Remove a variable from the binary quadratic model.
+
+        Returns:
+            hashable: The last variable added to the binary quadratic model.
+
+        Raises:
+            ValueError: If the binary quadratic model is empty.
+
+        """
+        if len(self._adj) == 0:
+            raise ValueError("pop from empty binary quadratic model")
+
+        v, neighbourhood = self._adj.popitem(last=True)
+
+        for u in neighbourhood:
+            if u != v:
+                self._adj[u].pop(v)
+
+        return v
+
+    def remove_interaction(self, u, v):
+        """Remove the interaction between variables u and v.
+
+        Args:
+            u (hashable):
+                A variable in the binary quadratic model.
+
+            v (hashable):
+                A variable in the binary quadratic model.
+
+        Returns:
+            bool: If there was an interaction to remove.
+
+        Raises:
+            ValueError: If either u or v is not a variable in the binary
+            quadratic model.
+
+        """
+        try:
+            self._adj[u].pop(v)
+        except KeyError:
+            # nothing pop
+            return False
+
+        # want to raise an exception in the case that they got out of sync
+        # as a sanity check
+        self._adj[v].pop(u)
```

Comment:
How come you don't need to `self._adj[u].pop(v)`?

Reply:
Happens above in the try-catch. We only get to this line if that one has occured